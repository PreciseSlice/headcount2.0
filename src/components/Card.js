import React from 'react';

const Card = (props) => {
  return (
    <article>
      <h3>{props.data.location}</h3>
      <p></p>
    </article>
  )
}
export default Card;