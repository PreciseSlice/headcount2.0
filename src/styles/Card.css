article {
  background-color: white;
  box-shadow: 1px 1px 5px lightgray;
  display: flex;
  flex-direction: column;
  height: 20rem;
  margin: 1rem;
  padding: .5rem;
  width: 12rem;
}

h3 {
  align-self: center;
  background-color: #5B5BE8;
  color: ghostwhite;
  font-family: 'Oswald', sans-serif;
  font-size: 1rem;
  padding: 0px 5px;
  width: 90%;
}

.element-wrapper {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding-left: 2rem;
}

.years {
  color: #1010A4;
}

.percents {

}