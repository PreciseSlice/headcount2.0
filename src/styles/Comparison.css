.Comparison {
  background-color: #E85B5B;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
}

.average {
  background-color: #E85B5B;
}

.Comparison h5 {
  padding: 0px;
}