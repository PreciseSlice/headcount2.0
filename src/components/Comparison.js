import React from 'react';
import Card from './Card';
import PropTypes from 'prop-types';

const Comparison = ( { selected }) => {
  const districtCards;
  if ( selected.length = 2) {
    districtCards = selected.map((district, index) => {
      <Card 
      data={district} 
      key={index}  />
    })
  }

  return ( 
    <div className="Comparison"> 
      Comparison component
      { districtCards }
   </div>
 )
}

export default Comparison;