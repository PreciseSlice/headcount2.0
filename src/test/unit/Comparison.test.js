/* eslint-disable */
import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount, render } from "enzyme";
import Comparison from "../../components/Comparison";
import { selected, containerData, cardData } from './mockData';
import Card from "../../components/Card";


const getComparison = function () {
  return {'COLORADO': 0.53, 'ACADEMY 20': 0.407, 'compared': 1.302}
}


describe("Comparison", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Comparison data={containerData} selected={selected} getComparison={getComparison} />);
  });

  it("should exist", () => {
    expect(wrapper).toBeDefined();
  });

  // it('should make 2 cards if 2 cards are passed into it', () => {
  //   expect(wrapper.find('Card').length).toEqual(2)
  // })

  // should render a comparision card
});
