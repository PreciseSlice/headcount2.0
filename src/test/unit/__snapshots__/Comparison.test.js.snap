// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comparison should render an average card in the middle of two Card components 1`] = `
<div
  className="Comparison"
>
  <Card
    clickCard={[Function]}
    data={
      Object {
        "data": Object {
          "2004": 0.24,
          "2005": 0.278,
          "2006": 0.337,
          "2007": 0.395,
          "2008": 0.536,
          "2009": 0.598,
          "2010": 0.64,
          "2011": 0.672,
          "2012": 0.695,
          "2013": 0.703,
          "2014": 0.741,
        },
        "location": "COLORADO",
      }
    }
    key="COLORADO"
    selected={
      Array [
        Object {
          "data": Object {
            "2004": 0.24,
            "2005": 0.278,
            "2006": 0.337,
            "2007": 0.395,
            "2008": 0.536,
            "2009": 0.598,
            "2010": 0.64,
            "2011": 0.672,
            "2012": 0.695,
            "2013": 0.703,
            "2014": 0.741,
          },
          "location": "COLORADO",
        },
        Object {
          "data": Object {
            "2004": 0.069,
            "2005": 0.509,
            "2006": 0.638,
            "2007": 0.994,
            "2008": 0.992,
            "2009": 1,
            "2010": 0.993,
            "2011": 0.994,
            "2012": 0.993,
            "2013": 0.989,
            "2014": 0.994,
          },
          "location": "COLORADO SPRINGS 11",
        },
      ]
    }
  />
  <article
    className="average-card"
  >
    <h5>
      Selected district 10-year averages:
    </h5>
    <h3>
      COLORADO
    </h3>
    <h5>
      53.0
      %
    </h5>
    <h3
      className="average"
    >
      RATIO
    </h3>
    <h5
      className="compared"
    >
      1.302
    </h5>
    <h3>
      ACADEMY 20
    </h3>
    <h5>
      40.7
      %
    </h5>
  </article>
  <Card
    clickCard={[Function]}
    data={
      Object {
        "data": Object {
          "2004": 0.069,
          "2005": 0.509,
          "2006": 0.638,
          "2007": 0.994,
          "2008": 0.992,
          "2009": 1,
          "2010": 0.993,
          "2011": 0.994,
          "2012": 0.993,
          "2013": 0.989,
          "2014": 0.994,
        },
        "location": "COLORADO SPRINGS 11",
      }
    }
    key="COLORADO SPRINGS 11"
    selected={
      Array [
        Object {
          "data": Object {
            "2004": 0.24,
            "2005": 0.278,
            "2006": 0.337,
            "2007": 0.395,
            "2008": 0.536,
            "2009": 0.598,
            "2010": 0.64,
            "2011": 0.672,
            "2012": 0.695,
            "2013": 0.703,
            "2014": 0.741,
          },
          "location": "COLORADO",
        },
        Object {
          "data": Object {
            "2004": 0.069,
            "2005": 0.509,
            "2006": 0.638,
            "2007": 0.994,
            "2008": 0.992,
            "2009": 1,
            "2010": 0.993,
            "2011": 0.994,
            "2012": 0.993,
            "2013": 0.989,
            "2014": 0.994,
          },
          "location": "COLORADO SPRINGS 11",
        },
      ]
    }
  />
</div>
`;
